doc/source/_static/gallery.html

```diff
@@ -0,0 +1,315 @@
+      L.tileLayer(data["url"], data).addTo(sampleMap);
+      tbl1 = document.createElement("table");
+      tbl1.className = "table-container";
+
+      for (let i = 0; i < nbOfRows; i++) {
+        const tr1 = tbl1.insertRow();
+        tr1.className = "line-container";
+        for (let j = 0; j < 2; j++) {
+          if (i === nbOfRows - 2 && j === 2) {
+            break;
+          } else {
+            const td1 = tr1.insertCell();
+            if (j == 0) {
+              // First column of the table : the one with the keys
+              td1.className = "key-cell";
+              td1.textContent = key[i];
+            } else {
+              // Second column of the table : the one with the values of the metadata
+              td1.className = "val-cell";
+              td1.textContent = val[i];
+            }
+          }
+        }
+      }
+      mainContainer.appendChild(tbl1);
+    } else {
+      // Case with apikey
+      var dict = accessData[basemap];
+      var keyString = dict["keyString"];
+
+      tbl2 = document.createElement("table");
+      tbl2.className = "table-container";
+      for (let i = 0; i < nbOfRows; i++) {
+        const tr2 = tbl2.insertRow();
+        tr2.className = "line-container";
+
+        for (let j = 0; j < 2; j++) {
+          if (i === nbOfRows - 2 && j === 2) {
+            break;
+          } else {
+            const td2 = tr2.insertCell();
+
+            if (j == 0) {
+              // First column of the table containing the keys of the metadata
+              td2.className = "key-cell";
+              td2.textContent = key[i];
+            } else {
+              // Second column of the table containing the values of the metadata
+              td2.className = "val-cell";
+
+              // create a single input and a button with onclick function for apikey
+              if (key[i] === keyString) {
+                var keyInput = document.createElement("input");
+                keyInput.type = "password";
+                keyInput.placeholder = "Enter your APIkey please";
+                keyInput.className = "key-container";
+                td2.append(keyInput);
+
+                var validationButton = document.createElement("button");
+                validationButton.className = "button-container";
+                td2.append(validationButton);
+                validationButton.innerHTML = "validate";
+                validationButton.onclick = get_keyCode;
+
+                function get_keyCode() {
+                  val[i] = keyInput.value;
+                  data[keyString] = keyInput.value;
+                  L.tileLayer(data["url"], data).addTo(sampleMap);
+                }
+              } else {
+                td2.textContent = val[i];
+              }
+            }
+          }
+        }
+      }
+      mainContainer.appendChild(tbl2);
+    }
+  } catch {}
+}
+
+getJSON(
+  "https://raw.githubusercontent.com/geopandas/xyzservices/main/xyzservices/data/providers.json",
```

Comment:
This can cause a bit of a trouble. Our documentation has two versions, one (default) pointing to the latest release, not the `main` branch, and the other one using the latest version of the main. This hard-coded approach only works for the latter. If we could fetch it from the `shared` folder where pip installs the JSON, it would be much better.

Reply:
Hi ! Many thanks for the review. First issue with iframe should now be resolved with the last commit. Concerning the second comment, I am not to understand how to address it.

Reply:
I have replaced the url by the access path to the .json file.